near/contracts/bridge/src/lib.rs

```diff
@@ -795,15 +880,65 @@ impl FastBridge {
     /// * The caller does not have any balance.
     #[payable]
     #[pause(except(roles(Role::UnrestrictedWithdraw)))]
-    pub fn withdraw(&mut self, token_id: AccountId, amount: Option<U128>) -> PromiseOrValue<U128> {
-        let recipient_id = env::predecessor_account_id();
-        let user_balance = self.get_user_balance(&recipient_id, &token_id);
+    pub fn withdraw(
+        &mut self,
+        token_id: AccountId,
+        amount: Option<U128>,
+        recipient_id: Option<AccountId>,
+        msg: Option<String>,
+    ) -> Promise {
+        let sender_id = env::predecessor_account_id();
+        self.withdraw_internal(token_id, amount, sender_id, recipient_id, msg)
+    }
+
+    fn withdraw_internal(
+        &mut self,
+        token_id: AccountId,
+        amount: Option<U128>,
+        sender_id: AccountId,
+        recipient_id: Option<AccountId>,
+        msg: Option<String>,
+    ) -> Promise {
+        let user_balance = self.get_user_balance(&sender_id, &token_id);
```

Comment:
What if the unlock were done for other user, not for `sender_id`? This is strange behavior when we unlock for one user and withdraw tokens for another.

Reply:
I didn't get you.
in the `unlock_and_withdraw`, we use the same `sender_id`

Reply:
Good catch; we have 4 solutions for this problem.
* Re-add the sender_id verification so the unlocking can only be done when the sender is the transfer originator.
* Prevent unlock and withdraw if the `sender_id != transfer originator`
* Allow unlock and withdraw just on `aurora_sender != None`
* Allow withdrawal by another account, but set the  `transfer originator` as `sender_id` on withdraw.

Reply:
I think the solution "Prevent unlock and withdraw if the sender_id != transfer originator" is the most logical.

Reply:
3f52745f3e663effb6b96dc490fdb09bf68213c0